controllers/release/release_adapter.go

```diff
@@ -315,7 +316,7 @@ func (a *Adapter) createReleasePipelineRun(releaseStrategy *v1alpha1.ReleaseStra
 	applicationSnapshot *applicationapiv1alpha1.ApplicationSnapshot) (*v1beta1.PipelineRun, error) {
 	pipelineRun := tekton.NewReleasePipelineRun("release-pipelinerun", releaseStrategy.Namespace).
 		WithOwner(a.release).
-		WithReleaseAndApplicationLabels(a.release.Name, a.release.Namespace, a.release.GetAnnotations()[logicalcluster.AnnotationKey], applicationSnapshot.Spec.Application).
+		WithReleaseAndApplicationMetadata(a.release.Name, a.release.Namespace, a.release.GetAnnotations()[logicalcluster.AnnotationKey], applicationSnapshot.Spec.Application, a.release, metadata.NewMetaCopy()).
```

Comment:
```suggestion
		WithReleaseAndApplicationMetadata(a.release, applicationSnapshot.Spec.Application).
```

Reply:
I was hoping you might mention this, because I was thinking the same thing. But I didn't want to intrude into this function signature too much, so I was simply appending arguments. I agree that collapsing the common args to a single `Release` instance makes more sense, and honestly I'm surprised it wasn't already that way before. The only weird part is dealing with the `logicalcluster` bits inside `WithReleaseAndApplicationMetadata`, as I'm unsure if that violates any sense of separation, compartmentalization, or whatever... I'll get this added in the next refactor today. :+1: